python/pyspark/ml/dl_util.py

```diff
@@ -0,0 +1,142 @@
+import os
+import tempfile
+import textwrap
+from typing import Any, Callable
+
+from pyspark import cloudpickle
+
+
+class FunctionPickler:
+    """
+    This class provides a way to pickle a function and its arguments.
+    It also provides a way to create a script that can run a
+    function with arguments if they have them pickled to a file.
+    It also provides a way of extracting the conents of a pickle file.
+    """
+
+    @staticmethod
+    def pickle_fn_and_save(
+        fn: Callable, file_path: str, save_dir: str, *args: Any, **kwargs: Any
+    ) -> str:
+        """
+        Given a function and args, this function will pickle them to a file.
+
+        Parameters
+        ----------
+        fn: Callable
+            The picklable function that will be pickled to a file.
+
```

Comment:
I think we should remove a space between each prameters per https://numpydoc.readthedocs.io/en/latest/format.html#parameters

Reply:
Done